database/buildSchema/45_activity_log.sql

```diff
@@ -125,6 +126,31 @@ CREATE TRIGGER outcome_update_activity_trigger
     WHEN (NEW.outcome <> OLD.outcome)
     EXECUTE FUNCTION outcome_activity_log ();
 
+-- SCHEDULED EVENT (Deadline) changes
+CREATE OR REPLACE FUNCTION public.deadline_extension_activity_log ()
+    RETURNS TRIGGER
+    AS $application_event$
+BEGIN
+    INSERT INTO public.activity_log (type, value, application_id, "table", record_id, details)
+        VALUES ('EXTENSION', NEW.event_code, NEW.application_id, TG_TABLE_NAME, NEW.id, json_build_object('newDeadline', NEW.time_scheduled, 'extendedBy', json_build_object('userId', NEW.editor_user_id, 'name', (
+                        SELECT
+                            full_name
+                        FROM "user"
+                        WHERE
+                            id = NEW.editor_user_id))));
+    RETURN NULL;
+END;
+$application_event$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS deadline_extension_activity_trigger ON public.application;
+
+CREATE TRIGGER deadline_extension_activity_trigger
+    AFTER UPDATE ON public.trigger_schedule
+    FOR EACH ROW
+    WHEN (NEW.time_scheduled > OLD.time_scheduled AND NEW.event_code = 'applicantDeadline')
```

Comment:
would this fire when time_scheduled is updated by subsequent LOQ ?

Reply:
Okay, yeah good point. I guess if I add the activity_log addition to the extendApplicationRoute like you suggested below, that'll take care of this problem too.

Reply:
Okay, this seems to work to handle that case: d184f5c3

(Using the "editor_user_id" column more effectively)